confidant/routes/saml.py

```diff
@@ -0,0 +1,61 @@
+import logging
+
+import flask
+from flask import jsonify, request, session
+
+from confidant import authnz
+from confidant.app import app
+
+
+@app.route('/v1/saml/metadata', methods=['GET'])
+def get_saml_metadata():
+    """
+    Generate SAML metadata XML describing the service endpoints.
+    """
+    return authnz.user_mod.generate_metadata()
+
+
+@app.route('/v1/saml/consume', methods=['POST'])
+def consume_saml_assertion():
+    """
+    The SAML attribute consumer service receives POST callbacks from the IdP.
+    """
+    return authnz.user_mod.consume_saml_assertion()
+
+
+@app.route('/v1/saml/login', methods=['GET'])
+def generate_saml_login_redirect():
+    """
+    Redirect to the SAML login page. You don't normally need to hit this
+    since any page with @authnz.require_auth will redirect to login.
+    """
+    return flask.redirect(
+        authnz.user_mod.login_redirect_url(return_to='/v1/saml/debug'))
```

Comment:
Yeah it's a little weird to leave that in there, since I was using it for debugging the IdP integration. But the reason this endpoint exists is for debugging SAML. You never hit this page unless you type this URL explicitly. When tracing the HTTP requests it's nice to have separate start and end pages — a bit confusing to start and end at the same page the way the `@require_auth` wrapped endpoints do.